File: viikko-1/Repeat.java
public class Repeat {
    public int find(String s) {
        return 1;
    }

    public static void main(String[] args) {
        Repeat r = new Repeat();
        System.out.println(r.find("aaa")); // 1
        System.out.println(r.find("abcd")); // 4
        System.out.println(r.find("abcabcabcabc")); // 3
        System.out.println(r.find("aybabtuaybabtu")); // 7
        System.out.println(r.find("abcabca")); // 7
    }
}

File: viikko-1/repeat.py
import re
def find(s):
    regex = ""
    for i in range(0, len(s)):
        regex += s[i]
        print(regex)
        if re.match(f"({regex})", s):
            return i + 1

    return len(s)
            

if __name__ == "__main__":
    print(find("aaa")) # 1
    print(find("abcd")) # 4
    print(find("abcabcabcabc")) # 3
    print(find("ybabtuaybabtu")) # 7
    print(find("abcabca")) # 7